forge/src/runner.rs

```diff
@@ -210,251 +186,234 @@ impl<'a, DB: DatabaseRef + Send + Sync> ContractRunner<'a, DB> {
         })
     }
 
-    #[tracing::instrument(name = "fuzz-test", skip_all, fields(name = %func.signature()))]
+    #[tracing::instrument(name = "fuzz-test", skip_all, fields(name = %func.signature(), %should_fail))]
     pub fn run_fuzz_test(
         &self,
         func: &Function,
+        should_fail: bool,
         runner: TestRunner,
-        _known_contracts: Option<&BTreeMap<String, (Abi, Vec<u8>)>>,
-        address: Address,
-        mut logs: Vec<RawLog>,
+        mut setup: TestSetup,
     ) -> Result<TestResult> {
-        // We do not trace in fuzz tests as it is a big performance hit
+        // Run fuzz test
         let start = Instant::now();
-        // TODO: Dedupe (`TestDescriptor`?)
-        let should_fail = func.name.starts_with("testFail");
-
-        let identified_contracts: Option<BTreeMap<Address, (String, Abi)>> = None;
-
-        // Wrap the executor in a fuzzed version
-        // TODO: When tracing is ported, we should disable it here.
-        let executor = FuzzedExecutor::new(&self.executor, runner, self.sender);
-        let FuzzTestResult { cases, test_error } =
-            executor.fuzz(func, address, should_fail, Some(self.contract));
-
-        // Rerun the failed fuzz case to get more information like traces and logs
-        if let Some(FuzzError { test_error: TestError::Fail(_, ref calldata), .. }) = test_error {
-            // TODO: When tracing is ported, we should re-enable it here to get traces.
-            let RawCallResult { logs: execution_logs, .. } =
-                self.executor.call_raw(self.sender, address, calldata.0.clone(), 0.into())?;
-            logs.extend(execution_logs);
-        }
-
-        let (success, counterexample, reason) = match test_error {
-            Some(err) => {
-                let (counterexample, reason) = match err.test_error {
-                    TestError::Abort(r) if r == "Too many global rejects".into() => {
-                        (None, Some(r.message().to_string()))
-                    }
-                    TestError::Fail(_, calldata) => {
-                        // Skip the function selector when decoding
-                        let args = func.decode_input(&calldata.as_ref()[4..])?;
-
-                        (Some(CounterExample { calldata, args }), None)
-                    }
-                    e => panic!("Unexpected test error: {:?}", e),
-                };
+        let mut result = FuzzedExecutor::new(&self.executor, runner, self.sender).fuzz(
+            func,
+            setup.address,
+            should_fail,
+            self.errors,
+        );
 
-                if !err.revert_reason.is_empty() {
-                    (false, counterexample, Some(err.revert_reason))
-                } else {
-                    (false, counterexample, reason)
-                }
-            }
-            _ => (true, None, None),
-        };
+        // Record logs, labels and traces
+        setup.logs.append(&mut result.logs);
```

Comment:
Yes, since they are part of the struct we can't move them out using `.extend` but I guess it would be possible if we destructure as per your previous comment